### ISSUE TITLE
Since version 1.12.0 my callback is not called

### ISSUE THREAD

**Issue by stavv** [BranchMetrics/android-branch-deep-linking#246]
I'm initializing Branch like this:
```
branch.initSession(new Branch.BranchReferralInitListener() {
            @Override
            public void onInitFinished(JSONObject referringParams, BranchError error) {
                if (error == null) {
                    String promoCode = getString(referringParams, PROMO_CODE_BRANCH_PARAM);
                    PromoUtil.storePromoCodeIfValid(promoCode);
                }

                listener.branchInitialized();
            }
        }, activity.getIntent().getData(), activity);

```
Since the dependency in the app is using a '+' in the end I got the latest version you released yesterday and since then my app behaves strange. After I changed it to be explicitly 1.11.3 everything worked fine.

When stepping into the code I noticed that the WeakReference to the callback becomes empty no matter what I did (even storing my callback in a global static variable).

Any ideas?

**Comment by sojanpr** [comment]
@stavv  We are sorry that you face this issue. We are working on it and update you asap

**Comment by iantheparker** [comment]
Aha! I spent all day on this, and I'm grateful that I'm not alone. I thought I was crazy.

**Comment by sojanpr** [comment]
@iantheparker Sorry for this issue. Unfortunately we are not able to reproduce this issue on our testing apps. So could you please  help us  a little bit to understand what causes it. 
It would be great if our support team can reach your for a quick help. Could you please write  to our awesome support team( support@branch.io  )and we can have  a 1:1 from there

**Comment by iantheparker** [comment]
Will do. But I don't have anything to add past what @stavv said. Everything was working great for months, and today after updating the version the callback in initSession just isn't getting hit.

**Comment by sojanpr** [comment]
@iantheparker Thanks for the update. Lets take care of it in support

**Comment by sojanpr** [comment]
@iantheparker @stavv We have figured out a case where the callback can get nullified. This is happening when`initSession` is called  before a branch auto session is triggered from life cycle method under certain circumstance.  Fix are added  to this PR.
https://github.com/BranchMetrics/android-branch-deep-linking/pull/247
 Do you think this will solve the issue?
Thank you very much for your help

**Comment by sojanpr** [comment]
@iantheparker @stavv We have  released SDK version 1.12.0 with the fix for the issue. Would you mind updating this issue after trying the new release

**Comment by stavv** [comment]
@sojanpr Just got back to work and it seems to work fine. Thanks for the quick response and solution (:

**Comment by sojanpr** [comment]
Thank you @stavv for the update. @iantheparker could you please close this issue if the fi work for you too

**Comment by iantheparker** [comment]
@stavv the callback is working! But now onto a new issue- I haven't been able to make a successful connection yet in 1.12.1. "Trouble initializing Branch.  Request to Branch server timed out. Please check your internet connectivity". I'll keep poking around, but if it's not clear soon I'll post a new issue.

**Comment by sojanpr** [comment]
@iantheparker  do you still have the issue of connection failure. We had a temporary  issue causing it and fixed on our servers couple of days ago. Could you please write to our awesome support team for a quick response (support@branch.io). We can take care of the issue 1:1 there. 
Thanks

**Comment by sojanpr** [comment]
@iantheparker @stavv We have found one more really corner case that can contribute this case. The fix has already released with v1.13.1. Thanks for your help
